snakemake/api.py

```diff
@@ -523,12 +523,6 @@ def execute_workflow(
                 "For local execution, --shared-fs-usage has to be unrestricted."
             )
 
-        self.snakemake_api.setup_logger(
-            stdout=executor_plugin.common_settings.dryrun_exec,
-            mode=self.workflow_api.workflow_settings.exec_mode,
-            dryrun=executor_plugin.common_settings.dryrun_exec,
-        )
```

Comment:
The reason for this extra call is that with calling execute, the dryrun info might be new, as well as the exec mode. One could of course offer a method to just update these in the logger api.

Reply:
Hmm, I'm not sure I follow. It seems setup logger is called here https://github.com/snakemake/snakemake/blob/8a80bdaed0d8524dffbdce71dd07eb123f71726d/snakemake/api.py#L141
How can the the info be new by time execute is called? Regardless, I think you're right, we can provide setters for these in the logger

Reply:
Could we just do this instead of calling setup_logger()?

```suggestion
         logger.stdout = executor_plugin.common_settings.dryrun_exec
         logger.mode = self.workflow_api.workflow_settings.exec_mode
         logger.dryrun = executor_plugin.common_settings.dryrun_exec
```

Reply:
Yes, I think so.